src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -37,7 +37,7 @@ public function validate($value, Constraint $constraint)
         $valid = filter_var($value, FILTER_VALIDATE_EMAIL);
 
         if ($valid) {
-            $host = substr($value, strpos($value, '@') + 1);
+            $host = $this->extractHostname($value);
```

Comment:
Since `extractHostname()` is used only once its body should be inlined here (there is no need for an additional method call).